packages/react-components/react-select/src/components/Select/Select.types.ts

```diff
@@ -40,3 +46,11 @@ export type SelectProps = Omit<ComponentProps<Partial<SelectSlots>, 'select'>, '
 };
 
 export type SelectState = ComponentState<SelectSlots> & Required<Pick<SelectProps, 'appearance' | 'inline' | 'size'>>;
+
+/**
+ * Data passed to the `onChange` callback when a new option is selected
+ */
+export type SelectOnChangeData = {
+  /* Updated `<select>` value, taken from either the selected option's value prop or inner text */
```

Comment:
_nit:_ Missing period. It also should probably have `/**` so that it is correctly picked up.

```suggestion
  /**
   * Updated `<select>` value, taken from either the selected option's value prop or inner text.
   */
```